src/ws/cockpitwebservice.c

```diff
@@ -1597,12 +1600,18 @@ create_web_socket_server_for_stream (const gchar **protocols,
                          host ? host : "localhost",
                          query ? "?" : "",
                          query ? query : "");
-  origin = g_strdup_printf ("%s://%s", secure ? "https" : "http", host);
 
-  connection = web_socket_server_new_for_stream (url, origin, protocols,
-                                                 io_stream, headers,
-                                                 input_buffer);
-  g_free (origin);
+  origins = cockpit_conf_strv ("WebService", "Origins", ' ');
+  if (origins == NULL)
```

Comment:
We may need to specify the full origins in the configuration file. What do you think?